pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -566,3 +566,11 @@ def test_setitem_boolean_mask(self, mask_type, float_frame):
         expected = df.copy()
         expected.values[np.array(mask)] = np.nan
         tm.assert_frame_equal(result, expected)
+
+    def test_setitem_boolean_mask_aligning(self):
```

Comment:
can you parameterize over .loc & setitem

Reply:
Done